ouroboros-consensus/src/Ouroboros/Storage/FS/API.hs

```diff
@@ -136,36 +138,34 @@ hGetExactly :: forall m h
 hGetExactly hasFS h n = go n []
   where
     go :: Int -> [BS.ByteString] -> m BL.ByteString
-    go 0 bss = return $ BL.fromChunks $ reverse bss
-    go remainingBytes acc = do
-      bs <- hGetSome hasFS h remainingBytes
-      if BS.null bs then do
-        path <- handlePath hasFS h
-        throwError (hasFsErr hasFS) FsError {
-            fsErrorType   = FsReachedEOF
-          , fsErrorPath   = path
-          , fsErrorString = "hGetExactly found eof before reading " ++ show n ++ " bytes"
-          , fsErrorStack  = callStack
-          , fsLimitation  = False
-          }
-      else go (remainingBytes - BS.length bs) (bs : acc)
-
--- | Like hGetExactly, but it does not throw an exception if eof is found.
-hGetLenient :: forall m h
-            . (HasCallStack, Monad m)
-            => HasFS m h
-            -> h
-            -> Int
-            -> m BL.ByteString
-hGetLenient hasFS h n = go n []
+    go remainingBytes acc
+      | remainingBytes <= 0 = return $ BL.fromChunks $ reverse acc
```

Comment:
.. which would then mean that this should be `==`, no` <=`. Feel free to ignore though :)